File: stage_5/SparkX/backend/app/models/entities/Task.go
package entities

import "time"

type Task struct {
	ID          int64     `json:"ID"`          // идентификатор
	Author      *Employee `json:"author"`      // Создатель задачи
	Executor    *Employee `json:"executor"`    // Исполнитель задачи
	Projects    *Project  `json:"project"`     // Проект к которому принадлежит задача
	States      string    `json:"states"`      // Состояние задачи
	Status      string    `json:"status"`      // Статус задачи
	Name        string    `json:"name"`        // Название задачи
	Description string    `json:"description"` // Описание задачи
	PlanHour    int64     `json:"planHour"`    // Планируемые часы на задачу
	ActualHour  int64     `json:"actualHour"`  // Актуальное время потраченное на задачу
	CreateTime  time.Time `json:"createTime"`  // Время создания задачи
}

File: stage_5/SparkX/frontend/src/js/src/models/entities/task.ts
import Employee from "./employee";
import Project from "./project";

export default class Task {
    public ID: number;
    public author: Employee;
    public executor: Employee;
    public project: Project;
    public states: string;
    public status: string;
    public name: string;
    public description: string;
    public planHour: number;
    public actualHour: number;
    public createTime: Date;

    constructor(
        id: number,
        author: Employee,
        executor: Employee,
        project: Project,
        states: string,
        status:  string,
        name: string,
        description: string,
        planHour: number,
        actualHour: number,
        createTime: string
    ) {
        this.ID = id;
        this.author = author;
        this.executor = executor;
        this.project = project;
        this.states = states;
        this.status = status;
        this.name = name;
        this.description = description;
        this.planHour = planHour;
        this.actualHour = actualHour;
        this.createTime = new Date(createTime);
    }
}